components/EntryModal.js

```diff
@@ -0,0 +1,154 @@
+import React, { useState } from "react";
+import {
+    View,
+    Modal,
+    StyleSheet,
+    Text,
+    TextInput,
+    Pressable,
+} from "react-native";
+
+const EntryModal = ({ onAdd, onCloseModal }) => {
+    //const [id, setId] = useState("");
+    const [date, setDate] = useState("");
+    const [item, setItem] = useState("");
+    const [cost, setCost] = useState("");
+    const [description, setDescription] = useState("");
+
+    const handleClose = () => {
+        onCloseModal(false);
+    };
+
+    const onSubmit = (e) => {
+        e.preventDefault();
+
+        if (!item) {
+            alert("Please add an entry");
+            return;
+        }
+
+        const entryToAdd = {
+            date,
+            item,
+            cost,
+            description,
+        };
+
+        onAdd(entryToAdd);
```

Comment:
i'm not sure i understand the potential issue. are you saying the resetting of the state in lines 39-42 might not run?

Reply:
Yeah, so if `onAdd` runs and as a part of that running it _closes_ `EntryModal` then `EntryModal` will be closed and therefore lines 38 and beyond in this `onSubmit` function would not run. See what I mean?

Reply:
Yes, I sure do. So I just commented out lines 39-42 and the code still works. Is it safe to say those lines were redundant, as well, since state becomes reinitialized when the modal opens?